airbyte-ci/connectors/pipelines/pipelines/tests/common.py

```diff
@@ -189,19 +204,57 @@ async def _run(self, connector_under_test_image_tar: Optional[File]) -> StepResu
         if not self.context.connector.acceptance_test_config:
             return StepResult(self, StepStatus.SKIPPED)
 
-        cat_container = await environments.with_connector_acceptance_test(self.context, connector_under_test_image_tar)
-        secret_dir = cat_container.directory("/test_input/secrets")
+        cat_container = await self._build_connector_acceptance_test(connector_under_test_image_tar)
+        cat_container = cat_container.with_(hacks.never_fail_exec(self.cat_command))
 
-        async with asyncer.create_task_group() as task_group:
-            soon_secret_files = task_group.soonify(secret_dir.entries)()
-            soon_cat_container_stdout = task_group.soonify(cat_container.stdout)()
+        step_result = await self.get_step_result(cat_container)
+        secret_dir = cat_container.directory(self.CONTAINER_SECRETS_DIRECTORY)
 
-        if secret_files := soon_secret_files.value:
+        if secret_files := await secret_dir.entries():
             for file_path in secret_files:
                 if file_path.startswith("updated_configurations"):
                     self.context.updated_secrets_dir = secret_dir
                     break
-        logs = soon_cat_container_stdout.value
-        if self.context.is_local:
-            await self.write_log_file(logs)
-        return self.pytest_logs_to_step_result(logs)
+        return step_result
+
+    def get_cache_buster(self) -> str:
+        """
+        This bursts the CAT cached results everyday.
+        It's cool because in case of a partially failing nightly build the connectors that already ran CAT won't re-run CAT.
+        We keep the guarantee that a CAT runs everyday.
+
+        Returns:
+            str: A string representing the current date.
+        """
+        return datetime.datetime.utcnow().strftime("%Y%m%d")
+
+    async def _build_connector_acceptance_test(self, connector_under_test_image_tar: File) -> Container:
```

Comment:
well done function.

But Im starting to notice we have a lot of connector code, that isnt obvious based on the file structure.

For example this is in test/common

should we start making it more specific. e.g.:
- test/connector_acceptance/common.py
- actions/connectors/tests.py

Reply:
The `test/common` module is originally made to receive connector testing step that are agnostic to the connector language. This is why `AcceptanceTest` belongs here.
The `actions` module was originally made to receive re-usable code. But I think I overused it and suffed action/environements with code that is not re-used in multiple context.